lib/beaker/ssh_connection.rb

```diff
@@ -66,20 +67,12 @@ def connect_block host, user, ssh_opts
     # connect to the host
     def connect
       #try three ways to connect to host (vmhostname, ip, hostname)
-      methods = []
-      if @vmhostname
-        @ssh ||= connect_block(@vmhostname, @user, @ssh_opts)
-        methods << "vmhostname (#{@vmhostname})"
+      # Try each method in turn until we succeed
+      methods = @ssh_connection_preference.dup
+      while @ssh.nil? && (not methods.empty?) do
+        @ssh ||= connect_block(instance_variable_get("@#{methods.shift}"), @user, @ssh_opts)
```

Comment:
@rishijavia `methods` will be empty here, using the original `@ssh_connection_preference` would be better.